### PR TITLE
Add wipe votes param to phrase uploader

### PR DESCRIPTION
**Pull request by LaurenceTews**

### CODE REVIEW

File: concord-server/server/src/main/java/com/rabidgremlin/concord/dao/UploadDao.java [hunk 1]
```diff
@@ -19,8 +19,12 @@
   @CreateSqlObject
   PhrasesDao phrasesDao();
 
-  @Transaction
-  default void uploadUnlabelledPhrases(List<UnlabelledPhrase> unlabelledPhrases)
+  enum EXISTING_VOTES
```
Comment (wilmol) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/dao/UploadDao.java]: rename to ```ExistingVotes``` ?

File: concord-server/server/src/main/java/com/rabidgremlin/concord/dao/UploadDao.java [hunk 2]
```diff
@@ -30,6 +34,19 @@ default void uploadUnlabelledPhrases(List<UnlabelledPhrase> unlabelledPhrases)
...
+      uploadUnlabelledPhrasesWithReplacement(batchedPhrases, phraseIds);
+    }
+    else
+    {
+      uploadUnlabelledPhrasesWithoutReplacement(batchedPhrases, phraseIds);
+    }
+  }
+
+  @Transaction
+  default void uploadUnlabelledPhrasesWithReplacement(List<UnlabelledPhrase> batchedPhrases, List<String> phraseIds)
```
Comment (wilmol) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/dao/UploadDao.java]: this method should be calling ```upsertBatch``` right?

File: concord-server/server/src/main/java/com/rabidgremlin/concord/resources/PhrasesResource.java [hunk 3]
```diff
@@ -130,12 +132,15 @@ public Response getNextPhraseToLabel(@ApiParam(hidden = true) @Auth Caller calle
...
-  public Response addPhrases(@ApiParam(hidden = true) @Auth Caller caller, UnlabelledPhrases unlabelledPhrases)
+  public Response addPhrases(@ApiParam(hidden = true) @Auth Caller caller, @DefaultValue("false") @QueryParam("replacevotes") String replaceVotes,
+    UnlabelledPhrases unlabelledPhrases)
   {
-    log.info("{} uploading {} phrases as json.", caller, unlabelledPhrases.getUnlabelledPhrases().size());
+    log.info("{} uploading {} phrases as json. Replace votes : {}", caller, unlabelledPhrases.getUnlabelledPhrases().size(), replaceVotes);
     log.debug("{}", unlabelledPhrases);
 
-    uploadDao.uploadUnlabelledPhrases(unlabelledPhrases.getUnlabelledPhrases());
+    UploadDao.EXISTING_VOTES voteContract = replaceVotes.equalsIgnoreCase("true") ? UploadDao.EXISTING_VOTES.REPLACE : UploadDao.EXISTING_VOTES.RETAIN;
```
Comment (wilmol) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/resources/PhrasesResource.java]: Put this logic in the enum class?

File: concord-server/server/src/main/java/com/rabidgremlin/concord/dao/UploadDao.java [hunk 4]
```diff
@@ -19,8 +19,21 @@
...
 
-  @Transaction
-  default void uploadUnlabelledPhrases(List<UnlabelledPhrase> unlabelledPhrases)
+  enum ExistingVotes
+  {
+    REPLACE, RETAIN;
+
+    public static ExistingVotes getVoteContract(String replaceVotes)
+    {
+      if (replaceVotes.equalsIgnoreCase("true"))
```
Comment (joedai) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/dao/UploadDao.java]: Use ternary operator to make it one liner

**Pull request merged**